detection_rules/attack.py

```diff
@@ -3,63 +3,114 @@
-    'Persistence': 'TA0003',
-    'Privilege Escalation': 'TA0004',
-    'Defense Evasion': 'TA0005',
-    'Credential Access': 'TA0006',
-    'Discovery': 'TA0007',
-    'Lateral Movement': 'TA0008',
-    'Execution': 'TA0002',
-    'Collection': 'TA0009',
-    'Exfiltration': 'TA0011',
-    'Command and Control': 'TA0010',
-    'Impact': 'TA0040'
-}
-TACTICS = list(TACTICS_MAP)
+
+import json
+import os
+import requests
+from collections import OrderedDict
+
+from .semver import Version
+from .utils import get_etc_path, get_etc_glob_path, read_gzip, gzip_compress
+
 PLATFORMS = ['Windows', 'macOS', 'Linux']
+tactics_map = {}
+
+
+def get_attack_file_path():
+    pattern = 'attack-v*.json.gz'
+    attack_file = get_etc_glob_path(pattern)
+    if len(attack_file) != 1:
+        raise FileNotFoundError(f'Missing required {pattern} file')
+    return attack_file[0]
+
 
-attack = load_etc_dump('attack.json')
+def load_attack_gz():
+    return json.loads(read_gzip(get_attack_file_path()))
+
+
+attack = load_attack_gz()
 
 technique_lookup = {}
+revoked = {}
 
 for item in attack["objects"]:
+    if item["type"] == "x-mitre-tactic":
+        tactics_map[item['name']] = item['external_references'][0]['external_id']
+
     if item["type"] == "attack-pattern" and item["external_references"][0]['source_name'] == 'mitre-attack':
         technique_id = item['external_references'][0]['external_id']
         technique_lookup[technique_id] = item
 
-matrix = {tactic: [] for tactic in TACTICS}
+        if item.get('revoked'):
+            revoked[technique_id] = item
+
+tactics = list(tactics_map)
+matrix = {tactic: [] for tactic in tactics}
+no_tactic = []
 attack_tm = 'ATT&CK\u2122'
 
 
 # Enumerate over the techniques and build the matrix back up
 for technique_id, technique in sorted(technique_lookup.items(), key=lambda kv: kv[1]['name'].lower()):
-    for platform in technique['x_mitre_platforms']:
-        if any(platform.startswith(p) for p in PLATFORMS):
-            break
-    else:
-        continue
-
-    for tactic in technique['kill_chain_phases']:
-        tactic_name = next(t for t in TACTICS if tactic['kill_chain_name'] == 'mitre-attack' and t.lower() == tactic['phase_name'].replace("-", " "))  # noqa: E501
-        matrix[tactic_name].append(technique_id)
+    kill_chain = technique.get('kill_chain_phases')
+    if kill_chain:
+        for tactic in kill_chain:
+            tactic_name = next(t for t in tactics if tactic['kill_chain_name'] == 'mitre-attack' and t.lower() == tactic['phase_name'].replace("-", " "))  # noqa: E501
+            matrix[tactic_name].append(technique_id)
+        else:
+            no_tactic.append(technique_id)
 
 for tactic in matrix:
     matrix[tactic].sort(key=lambda tid: technique_lookup[tid]['name'].lower())
 
 
-TECHNIQUES = {v['name'] for k, v in technique_lookup.items()}
+technique_lookup = OrderedDict(sorted(technique_lookup.items()))
+TECHNIQUES = sorted({v['name'] for k, v in technique_lookup.items()})
```

Comment:
`tactics` but `TECHNIQUES`?
don't care which, but should be consistent

Reply:
agree- will update